src/core/wee-network.c

```diff
@@ -1689,9 +1671,39 @@ network_connect_with_fork (struct t_hook *hook_connect)
             unhook (hook_connect);
             return;
         }
+        gnutls_certificate_allocate_credentials (&HOOK_CONNECT(hook_connect, gnutls_xcred));
         gnutls_credentials_set (*HOOK_CONNECT(hook_connect, gnutls_sess),
                                 GNUTLS_CRD_CERTIFICATE,
-                                gnutls_xcred);
+                                HOOK_CONNECT(hook_connect, gnutls_xcred));
+        network_set_gnutls_ca_file (HOOK_CONNECT(hook_connect, gnutls_xcred));
+        rc = (int) HOOK_CONNECT(hook_connect, gnutls_cb)
+            (hook_connect->callback_pointer,
+             hook_connect->callback_data,
+             *HOOK_CONNECT(hook_connect, gnutls_sess),
+             NULL, 0, NULL, 0, NULL,
+             WEECHAT_HOOK_CONNECT_GNUTLS_CB_INIT_XCRED);
```

Comment:
Maybe this isn't the best to call the GnuTLS callbacks function when simply initializing the trust list because strictly it isn't a callback for a GnuTLS function. `INIT_XCRED` could also be added as a case under the usual `hook_connect` callback if that's more appropriate, although to me it seems more related to using GnuTLS specifically and most connections will not need to touch that part.